File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/PublisherPicture.java
package com.amakakeru.mangaworld.Entity;

import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@Table(name = "publisher_picture")
public class PublisherPicture implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "pp_id")
    private Long ppId;

    @Lob
    @Column(name = "pp_picture", nullable = false)
    private byte[] ppPicture;

    @OneToOne
    @JoinColumn(name = "p_id", nullable = false, unique = true)
    private Publisher publisher;
}

File: Frontend/manga-world/src/Model/PublisherPicture.ts
import Publisher from "./Publisher";

class publisherPicture {
    ppId: number;
    ppPicture: string;
    publisher: Publisher;

    constructor(ppId: number, ppPicture: string, publisher: Publisher) {
        this.ppId = ppId;
        this.ppPicture = ppPicture;
        this.publisher = publisher;
    }
}

export default publisherPicture;